components/ContractStatus/ContractStatus.tsx

```diff
@@ -9,57 +12,45 @@ import { RatesNegative } from './RatesNegative';
 import { RatesPositive } from './RatesPositive';
 
 export function ContractStatus() {
-  const {
-    pricesData,
-    fetching: pricesDataFetching,
-    error: pricesDataError,
-  } = useControllerPricesData();
-  const contractAPR = useMemo(() => {
-    if (!pricesData || pricesData.targetValues.length < 2) {
+  const { target: currentTarget, lastUpdated, underlying } = useController();
+  const newTargetResult = useTarget();
+  const mark = useLatestMarketPrice();
+
+  const currentTargetNumber = useMemo(() => {
+    return parseFloat(
+      ethers.utils.formatUnits(currentTarget, underlying.decimals),
+    );
+  }, [currentTarget, underlying.decimals]);
+  const newTargetNumber = useMemo(() => {
+    if (!newTargetResult) {
       return null;
     }
-    const l = pricesData.targetValues.length;
-    const cur = pricesData.targetValues[l - 1];
-    const prev = pricesData.targetValues[l - 2];
-    const change = percentChange(prev.value, cur.value);
-    // convert to APR
-    return (change / (cur.time - prev.time)) * SECONDS_IN_A_YEAR;
-  }, [pricesData]);
+    return parseFloat(
+      ethers.utils.formatUnits(newTargetResult.newTarget, underlying.decimals),
+    );
+  }, [newTargetResult, underlying.decimals]);
 
-  const { mark, target } = useMemo(() => {
-    if (!pricesData) {
-      return {
-        mark: 0,
-        target: 0,
-      };
-    }
-    const { markValues, targetValues } = pricesData;
-    // This happens on a brand new controller that doesn't have data yet.
-    if (markValues.length === 0 || targetValues.length === 0) {
-      return {
-        mark: 0,
-        target: 0,
-      };
+  const contractAPR = useMemo(() => {
+    if (!newTargetResult || !newTargetNumber) {
+      return null;
     }
-    const mark = markValues[markValues.length - 1].value;
-    const target = targetValues[targetValues.length - 1].value;
-    return { mark, target };
-  }, [pricesData]);
+    const change = percentChange(currentTargetNumber, newTargetNumber);
+    // convert to APR
+    return (
+      (change / (newTargetResult.timestamp - lastUpdated)) * SECONDS_IN_A_YEAR
+    );
+  }, [newTargetNumber, lastUpdated, currentTargetNumber, newTargetResult]);
 
-  if (pricesDataFetching) {
+  if (!contractAPR || !mark) {
     return <Fieldset>Loading price data...</Fieldset>;
   }
 
-  if (pricesDataError || !contractAPR) {
```

Comment:
can't really have an error state anymore, since useLatestMarketPrice and useTarget don't return errors. maybe we should add those in a future PR